packages/react-native/ReactCommon/react/renderer/components/view/BaseViewProps.cpp

```diff
@@ -392,19 +392,33 @@ Transform BaseViewProps::resolveTransform(
     LayoutMetrics const& layoutMetrics) const {
   float viewWidth = layoutMetrics.frame.size.width;
   float viewHeight = layoutMetrics.frame.size.height;
-  if (!transformOrigin.isSet() || (viewWidth == 0 && viewHeight == 0)) {
-    return transform;
+  auto transformMatrix = Transform{};
+
+  if (viewWidth == 0 && viewHeight == 0){
+    return transformMatrix;
```

Comment:
Different from previous behaviour which returned the plain transform. Not sure it matters.

Reply:
True. Added this as an optimisation to avoid the operations loop as `FromTransformOperation` has become size dependent. I am also not sure transformation with zero dimensions matter though (with border-box). It's an assumption though.

At some point I would like to explore moving transform matrix generation to shadow view and native view can implement a method, similar to how `updateLayoutMetrics` works.